### PR TITLE
Fix NPE in DomainTranslator when coercing null values

### PR DESCRIPTION
**Pull request by erichwang**

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.138.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+=============
+Release 0.138
+=============
+
+General Changes
+---------------
+
+* Fix planning bug with null literal coercions.
```
Comment (cberner) [inline presto-docs/src/main/sphinx/release/release-0.138.rst]: put "null" in backticks and capitalize it?

**Comment by cberner** [comment]
Looks good. Do we need to worry about this being broken anywhere else?

File: presto-main/src/main/java/com/facebook/presto/sql/planner/DomainTranslator.java [hunk 2]
```diff
@@ -361,14 +361,25 @@ protected ExtractionResult visitComparisonExpression(ComparisonExpression node,
...
-            return createComparisonExtractionResult(normalized.getComparisonType(), symbol, fieldType, coerced, complement);
+        private Optional<NullableValue> coerce(NullableValue value, Type targetType)
+        {
+            if (!TypeRegistry.canCoerce(value.getType(), targetType)) {
+                return Optional.empty();
+            }
+            if (value.isNull()) {
+                return Optional.of(NullableValue.asNull(targetType));
+            }
+            Object coercedValue = new FunctionInvoker(metadata.getFunctionRegistry())
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/DomainTranslator.java]: We should make FunctionInvoker aware of `@nullable` argument annotations and short circuit if they are not present (per standard sql semantics)

**Comment by erichwang** [comment]
Fixes #4560

**Pull request closed without merging**